components/react/hooks/src/useControlledState/index.js

```diff
@@ -1,21 +1,20 @@
-import {useState, useEffect, useRef, useCallback} from 'react'
+import {useState, useLayoutEffect, useCallback} from 'react'
+import useMountedState from '../useMountedState'
 
 /**
  * React hook for combine the state of a value prop and its default value
  **/
 const useControlledState = (controlledValue, defaultValue) => {
-  const isFirst = useRef(true)
   const [initialValue] = useState(
     controlledValue === undefined ? defaultValue : controlledValue
   )
   const [value, setValue] = useState(initialValue)
-  useEffect(() => {
-    if (isFirst.current) {
-      isFirst.current = false
-    } else {
+  const isMounted = useMountedState()
+  useLayoutEffect(() => {
+    if (isMounted()) {
```

Comment:
I think that it changes behaviour. In changed code isFirst was set to true, then moved to false after "mount" and then `setValue`
Isn't it?
```suggestion
    if (isMounted() === false) {
```

Reply:
Its a semantic difference. isFirst means isNotMounted